### PR TITLE
Add technical overview

### PR DESCRIPTION
**Pull request by richardhuaaa**
Adds a technical overview describing XMTP v3. Thank you @jhaaaa and @jazzz for contributing to this document!

### CODE REVIEW

**Review by jazzz — APPROVED** [review]
Excited for this work to continue in the open. 👍

**Review by neekolas — APPROVED** [review]
:shipit:

File: TECHNICAL_OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+The first iteration is built on top of [Matrix’s Olm](https://gitlab.matrix.org/matrix-org/olm/-/blob/master/docs/olm.md), an implementation of the cryptographic ratchet described by Signal’s [Double Ratchet Algorithm](https://whispersystems.org/docs/specifications/doubleratchet/). This document describes the details of this system in the context of XMTP, a permissionless messaging network for blockchain identities.
+
+## Objectives
+
+- **Grow the XMTP network securely:** The suite of security properties introduced in XMTP v3 will reduce risk from potentially malicious apps and actors as we grow the XMTP network. These properties are displayed below.
+- **More transparent and agile protocol development:** The core of XMTP v3 is [libxmtp](https://www.notion.so/DRAFT-XMTP-v3-technical-design-17e63194ef7c4803b7eb3dd6051b6019?pvs=21), a Rust library that can be consumed across multiple platforms. Streamlining XMTP’s cryptographic code and protocol logic into one open-source repository allows for increased auditability, security, and development velocity.
```
Comment (jhaaaa) [inline TECHNICAL_OVERVIEW.md:25]:
```suggestion
- **More transparent and agile protocol development:** The core of XMTP v3 is [libxmtp](https://github.com/xmtp/libxmtp), a Rust library that can be consumed across multiple platforms. Streamlining XMTP’s cryptographic code and protocol logic into one open-source repository allows for increased auditability, security, and development velocity.
```

Comment (jhaaaa) [reply TECHNICAL_OVERVIEW.md:25]: @richardhuaaa Just one link tweak here! <3

File: TECHNICAL_OVERVIEW.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+Future versions of the protocol will build group messaging channels on top of the peer-to-peer messaging primitive introduced in v3 to mitigate these issues.
+
+**Exposed metadata**
+
+- **Contact bundles:** All contact bundles (and accompanying signature text) are visible to the XMTP network. This is by design and necessary for session-based encryption. Observers are able to deduce the following for a given wallet address:
+  - Number of messaging installations
+  - Creation time of each installation
+- **Payload metadata**: Payload metadata available on the network is limited to:
+  - Recipient installation address: Inbound topics contain a unique `installation_id`. for example, `message-0dYHO5xN+oj3T1gjC3odBj81j0ToP4l62N7oAFzrvGs`.
```
Comment (jhaaaa) [inline TECHNICAL_OVERVIEW.md:273]:
```suggestion
  - Recipient installation address: Inbound topics contain a unique `installation_id`. For example, `message-0dYHO5xN+oj3T1gjC3odBj81j0ToP4l62N7oAFzrvGs`.
```

**Review by jhaaaa — APPROVED** [review]
Thank you for this PR, @richardhuaaa - looking forward to all we'll learn and build together!

File: TECHNICAL_OVERVIEW.md [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+- An **account** consists of multiple app installations. To learn more, see [Account management](#account-management).
+- A **conversation** consists of multiple pair-wise sessions between installations. To learn more, see [Conversation management](#conversation-management).
+- Messages are individually encrypted and delivered at the **session** level. To learn more, see [Session management](#session-management).
+- Delivery of messages happens via network **topics**. To learn more, see [Network topic structure](#network-topic-structure).
+
+### Network topic structure
+
+XMTP nodes currently store data using Waku topics. Clients may publish, request, or subscribe to data from these topics. The purpose of these topics will be described in subsequent sections.
+
+- **`contact-<walletAddress>`**: Signed installation contact bundles for a given user account.
```
Comment (mkobetic) [inline TECHNICAL_OVERVIEW.md:62]: Is this going to interfere with v2?

Comment (neekolas) [reply TECHNICAL_OVERVIEW.md:62]: We use a different prefix for all v3 topics so there is no potential for collision.

**Comment by richardhuaaa** [comment]
For now, yes. XMTP v3 is still a work-in-progress and we will likely update this as we continue to make progress.

In future we may have separate, more detailed technical specs as needed.

**Pull request merged**